x-pack/plugins/spaces/server/spaces_client/spaces_client.ts

```diff
@@ -57,6 +63,12 @@ export interface ISpacesClient {
    * @param id the id of the space to delete.
    */
   delete(id: string): Promise<void>;
+
+  /**
+   * Disables the specified legacy URL aliases.
+   * @param aliases the aliases to disable.
+   */
+  disableLegacyUrlAliases(aliases: LegacyUrlAliasTarget[]): Promise<void>;
```

Comment:
Good question.
Aliases can only exist when Spaces is enabled (Aliases do not exist / function for the Default space). 

Since this is primarily a Spaces concern, it made sense to me to expose this from the Spaces plugin.

We could expose this from Core instead, if you want. But Spaces is the only consumer, so I'm not sure we should do that.

Reply:
I'm a little concerned about the isolation between core and spaces, given that the type is registered from core. But given my other 🙈 comments, we probably just need to accept that the feature is 'shared' between core and security (who said as always?) and keep it that way. I agree that expanding core's surface API for something only meant to be used from inside security doesn't make a lot of sense.